hyperon_das_atomdb/adapters/redis_mongo_db.py

```diff
@@ -615,14 +615,17 @@ def get_atom_as_dict(self, handle) -> dict:
                 answer["name"] = document["name"]
         return answer
 
-    def count_atoms(self) -> Tuple[int, int]:
-        nodes_count = self.mongo_atoms_collection.count_documents(
-            {FieldNames.COMPOSITE_TYPE: {'$exists': False}}
-        )
-        links_count = self.mongo_atoms_collection.count_documents(
-            {FieldNames.COMPOSITE_TYPE: {'$exists': True}}
-        )
-        return (nodes_count, links_count)
+    def count_atoms(self, parameters: Optional[Dict[str, Any]] = None) -> Tuple[int, int]:
```

Comment:
2 points

1 - Why does the parameter need to be a dict? why not just put the bool type?

```
def count_atoms(self, precise: bool = False) -> Tuple[int, int]:
if precise:
...
else:
...
```

2 - I understand why you return (node, link) when the 'precise' parameter is sent and (atoms, 0) otherwise, because we only have one collection. But it's a little strange to change the idea of ​​return. In one way I have Nodes and Links and in another I only have Atoms. You see? For me it's a little strange

My suggestion is to change the return type. for something clearer. Some like that:

```
def count_atoms(self, precise: bool = False) -> Dict[str, int]:
    if precise:
        nodes_count = self.mongo_atoms_collection.count_documents({FieldNames.COMPOSITE_TYPE: {'$exists': False}})
        links_count = self.mongo_atoms_collection.count_documents({FieldNames.COMPOSITE_TYPE: {'$exists': True}})
        return {'nodes': nodes_count, 'links': links_count}
    else:
        atoms_count = self.mongo_atoms_collection.estimated_document_count()
        return {'atoms': atoms_count}
```

But we're going to change the type and I'm not sure we can. What do you think @andre-senna ?

Reply:
I think (2) is a good idea.

And you have a point in (1) as well. Actually I was also thinking in making this comment but got distracted by other comments I posted in the Mattermost channel and forgot to mention it.

Reply:
1- There are other parameters for query-engine to use, but the precise key can be a boolean
2- LGTM, I'll change the return